index.js

```diff
@@ -11,63 +12,86 @@ const QUESTIONS = [{
+    // return back 200 status code to the client
 
-  //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
+    if(USERS.find(user=>user.username == req.body.username)){
 
+        return res.status(400).json({ error : 'User already exists'})
 
-  // return back 200 status code to the client
-  res.send('Hello World!')
-})
-
-app.post('/login', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
-
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
+    }else{
 
+    const user = {username: req.body.username, password: req.body.password}
+    USERS.push(user)
+    
+    res.status(200).json({ message: 'User registered successfully'})
 
-  // If the password is the same, return back 200 status code to the client
-  // Also send back a token (any random string will do for now)
-  // If the password is not the same, return back 401 status code to the client
-
-
-  res.send('Hello World from route 2!')
+    }
 })
 
-app.get('/questions', function(req, res) {
-
-  //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
+app.post('/login', (req, res)=>{
+
+    // Add logic to decode body
+    // body should have email and password
+
+    // Check if the user with the given email exists in the USERS array
+    // Also ensure that the password is the same
+    // If the password is the same, return back 200 status code to the client
+    // Also send back a token (any random string will do for now)
+    // If the password is not the same, return back 401 status code to the client
+
+    const user = USERS.find(user => user.username == req.body.username)
+  if(!user){
+    return res.status(401).json({error: 'Invalid user_name'})
+  }
+  //check password
+  if(user.password !== req.body.password){
+    return res.status(401).json({error: 'Invalid password'})
+  }
+  //send success message
+  res.json({ message: 'Login successful' });
 })
 
-app.get("/submissions", function(req, res) {
-   // return the users submissions for this problem
-  res.send("Hello World from route 4!")
-});
+app.get('/questions', (req, res)=>{
+    //return the user all the questions in the QUESTIONS array
+    
+    res.json(QUESTIONS)
+})
 
+app.get("/submissions", (req, res)=>{
+    res.json(SUBMISSION)
+})
 
-app.post("/submissions", function(req, res) {
-   // let the user submit a problem, randomly accept or reject the solution
+app.post("/submissions", (req, res)=>{
+    // let the user submit a problem, randomly accept or reject the solution
    // Store the submission in the SUBMISSION array above
-  res.send("Hello World from route 4!")
-});
-
-// leaving as hard todos
-// Create a route that lets an admin add a new problem
-// ensure that only admins can do that.
-
+   
```

Comment:
In General any website similar to leetcode will only let logged in users to submit solutions.

it's good to verify if the user exists in your datastore. storing submissions without checking identity of users will let anybody to post a submission without creating the account.